### PR TITLE
fix(bug): operations in the HikeDetailsScreen show a toast when there is no internet connection

### PR DESCRIPTION
**Pull request by mchac15**
# Summary
This PR adds a timeout to operations that are related to the `HikeDetailsScreen` with the `SavedHikesRepository` so that, whenever there is no internet connection the operations timeout and a Toast is shown on the screen.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -235,4 +235,5 @@
 
     <string name="delete_account_button_label">Delete Account</string>
     <string name="google_account_created_confirmation">Connected Google Account to your device successfully. Please wait while we retry the signup.</string>
+    <string name="generic_error_message_internet_connection">An error occurred. Check your internet connection</string>
```
Comment (davidholzwarth) [inline app/src/main/res/values/strings.xml:238]: Change: Here, perhaps warn the user that the behavior is now undefined. The change may be saved or not, we can not provide a guarantee for either.

**Review by davidholzwarth — CHANGES_REQUESTED** [review]
Please change the error string

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (mchac15) [reply app/src/main/res/values/strings.xml]: Fixed

**Review by davidholzwarth — APPROVED** [review]
LGTM

**Pull request merged**